R/gamlss-tidiers.R

```diff
@@ -1,39 +1,53 @@
-#' @templateVar class gamlss
-#' @template title_desc_tidy
-#'
-#' @param x A `gamlss` object returned from [gamlss::gamlss()].
-#' @template param_quick
-#' @template param_unused_dots
-#' 
-#' @evalRd return_tidy(
-#'   parameter = "Type of coefficient being estimated: `mu`, `sigma`,
-#'     `nu`, or `tau`.",
-#'   "term",
-#'   "estimate",
-#'   "std.error",
-#'   "statistic",
-#'   "p.value"
-#' )
-#' 
+#' Tidying methods for gamlss objects
+#'
+#' Tidying methods for "gamlss" objects from the gamlss package.
+#'
+#' @param x A "gamlss" object
+#' @param quick Whether to perform a fast version, and return only the coefficients
+#' @param conf.int whether to return confidence intervals
+#' @param ... arguments passed to \code{confint.gamlss}
+#'
+#' @name gamlss_tidiers
+#'
+#' @template boilerplate
+#'
+#' @return A tibble with one row for each coefficient, containing columns:
+#'   \item{parameter}{type of coefficient being estimated: \code{mu}, \code{sigma}, \code{nu}, or \code{tau}}
+#'   \item{term}{term in the model being estimated and tested}
+#'   \item{estimate}{estimated coefficient}
+#'   \item{std.error}{standard error}
+#'   \item{statistic}{t-statistic}
+#'   \item{p.value}{two-sided p-value}
+#'
 #' @examples
-#' 
-#' library(gamlss)
-#' 
-#' g <- gamlss(
-#'   y ~ pb(x),
-#'   sigma.fo = ~ pb(x),
-#'   family = BCT,
-#'   data = abdom,
-#'   method = mixed(1, 20)
-#' )
-#'
-#' tidy(g)
+#' \dontrun{
+#' if (requireNamespace("gamlss", quietly = TRUE) &&
+#'   requireNamespace("gamlss.data", quietly = TRUE)) {
+#'   data(abdom, package = "gamlss.data")
+#'
+#'   mod <- gamlss(y ~ pb(x),
+#'     sigma.fo = ~ pb(x), family = BCT,
+#'     data = abdom, method = mixed(1, 20)
+#'   )
+#' }
+#' ## load stored object
+#' mod <- readRDS(system.file("extdata", "gamlss_example.rds",
+#'   package = "broom.mixed"
+#' ))
+#' tidy(mod)
+#' }
 #'
 #' @export
-tidy.gamlss <- function(x, quick = FALSE, ...) {
+
+tidy.gamlss <- function(x, quick = FALSE, conf.int = FALSE, conf.level = 0.95, ...) {
   if (quick) {
     co <- stats::coef(x)
-    return(tibble(term = names(co), estimate = unname(co)))
+    return(dplyr::tibble(term = names(co), estimate = unname(co)))
+  }
+
+  # need gamlss for summary to work
+  if (!requireNamespace("gamlss", quietly = TRUE)) {
```

Comment:
I guess we can keep this but we don't do this anywhere else in `broom`. At some point I'll write a helper function that makes this a nice single function call.